iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/NetworkModificationList.java

```diff
@@ -36,4 +36,21 @@ public void apply(Network network, NamingStrategy namingStrategy, boolean throwE
                       ComputationManager computationManager, ReportNode reportNode) {
         modificationList.forEach(modification -> modification.apply(network, namingStrategy, throwException, computationManager, reportNode));
     }
+
+    @Override
+    public NetworkModificationImpact hasImpactOnNetwork(Network network) {
+        boolean hasImpact = false;
+        for (NetworkModification modification : modificationList) {
+            NetworkModificationImpact unitaryImpact = modification.hasImpactOnNetwork(network);
+            if (unitaryImpact == NetworkModificationImpact.CANNOT_BE_APPLIED) {
+                // If a sub-modification cannot be applied, the whole modification cannot be applied too
+                impact = unitaryImpact;
+                return impact;
+            } else if (hasImpact || unitaryImpact == NetworkModificationImpact.HAS_IMPACT_ON_NETWORK) {
+                hasImpact = true;
+            }
+        }
+        impact = hasImpact ? NetworkModificationImpact.HAS_IMPACT_ON_NETWORK : NetworkModificationImpact.NO_IMPACT_ON_NETWORK;
+        return impact;
```

Comment:
I think the intention could be clearer if we simplify the code a little bit. Something like:
```suggestion
        Set<NetworkModificationImpact> unitaryImpacts = modificationList.stream().map(m -> m.hasImpactOnNetwork(network)).distinct().toSet();
        if (unitaryImpacts.contains(CANNOT_BE_APPLIED)) {
            return CANNOT_BE_APPLIED;
        }
        if (unitaryImpacts.contains(HAS_IMPACT_ON_NETWORK)) {
            return HAS_IMPACT_ON_NETWORK;
        }
        return NO_IMPACT_ON_NETWORK;
```

Reply:
good point